modules/engine/src/model/model.ts

```diff
@@ -446,16 +446,14 @@ export class Model {
       )();
     }
     for (const [bufferName, buffer] of Object.entries(buffers)) {
-      const bufferLayout = this.bufferLayout.find(layout => layout.name === bufferName);
+      const bufferLayout = this.bufferLayout.find(layout => getAttributeNames(layout).includes(bufferName));
       if (!bufferLayout) {
         log.warn(`Model(${this.id}): Missing layout for buffer "${bufferName}".`)();
         continue; // eslint-disable-line no-continue
       }
 
       // For an interleaved attribute we may need to set multiple attributes
-      const attributeNames = bufferLayout.attributes
-        ? bufferLayout.attributes?.map(layout => layout.attribute)
-        : [bufferLayout.name];
+      const attributeNames = getAttributeNames(bufferLayout);
       let set = false;
       for (const attributeName of attributeNames) {
         const attributeInfo = this._attributeInfos[attributeName];
```

Comment:
Yes the application has to pass in a bufferLayout. It can optionally specify a shaderLayout.

shaderLayout used to be required in WebGPU, but now with the addition of wgsl_reflect, apps don't need to do that anymore on either WebGL or WebGPU.

So I would have thought that there would not be too much need to issue warnings, but perhaps you are seing reasons?

Reply:
Thanks @ibgreen! I'll take a closer look at the case that was producing this warning for me, and open a PR (with more details / explanation) if that still seems useful.